### ISSUE TITLE
Leaking issue in LegacyYouTubePlayerView

### ISSUE THREAD

**Issue by hoangthan** [PierfrancescoSoffritti/android-youtube-player#1058]
# Bug Report
Leaking issue in LegacyYouTubePlayerView

### Description of the Bug
In init block of LegacyYouTubePlayerView. You're adding the listener to NetworkBroadcastReceiver

`
 networkObserver.listeners.add(object : NetworkObserver.Listener {
      override fun onNetworkAvailable() {
        if (!isYouTubePlayerReady) {
          initialize()
        }
        else {
          playbackResumer.resume(webViewYouTubePlayer.youtubePlayer)
        }
      }

      override fun onNetworkUnavailable() { }
    })
`

It will fine if the view is bound with a lifecycle like fragment/activity. 
But in case the LegacyYouTubePlayerView is put on custom view which has no default lifecycle --> The listener will be leaked.

### Recommendation
Moving the block adding the network state listener from init to view.doOnAttach {  //Add network listener }
And remove it when view is going on detached by view.onViewDetached { //Remove network listener }

This will make all the user consistence and no need to care about the lifecycle of view.

Youtube Player Library Version: 

### Steps to reproduce the bug
---
name: Bug report
about: Use this template for reporting bugs
labels: bug
---

 

# Bug Report

### Environment details

Android Version: 
API Version: 

Tested devices:

Youtube Player Library Version: 

### Additional information
<!-- Provide here further information that might help understand the issue or reproduce the bug, e.g. 
- standalone code snippets of implementation that can copy-pasted and work without any missing classes
- screenshot(s) of visualising the bug
- error logs
- any useful background infromation

**Comment by hoangthan** [comment]
@PierfrancescoSoffritti Yes I know we should the release function to release the resource once the view is going on destroyed.
Since you're currently implementing LifeCycle to listen to the state of lifeCycleOwner to release resources once it going onDestroyed (if has any). So why do you make it run automatically to make the developer can avoid the leaking issue if forgets to bind it with the view lifecycleOwner ?
And the last one, if we put the YoutubePlayerView in a custom view, we still need call the release function when view is being on detached. Why don't you put it onDetached of YoutubePlayerView itself ?

**Comment by PierfrancescoSoffritti** [comment]
Ok, that seems reasonable. Feel free to send a pull request if you have time :)

**Comment by hoangthan** [comment]
Hi @PierfrancescoSoffritti 
I've submitted a PR: https://github.com/PierfrancescoSoffritti/android-youtube-player/pull/1059
